### PR TITLE
Implement the latest data scrape

### PR DESCRIPTION
**Pull request by decentral1se**
On the way to pulling down the data from [here](https://www.esrl.noaa.gov/gmd/webdata/ccgg/trends/co2_mlo_weekly.csv) and parsing it into a database model.

Hackity hack hacking.

Will close off https://github.com/giving-a-fuck-about-climate-change/carbondoomsday-django/issues/1.

### CODE REVIEW

**Comment by blueyed** [comment]
btw: no punctuation with commit log summaries is nicer.

File: setup.cfg [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [tool:pytest]
-addopts = --tb=short --ds=carbondoomsday.settings
+addopts = --tb=short --ds=carbondoomsday.settings --disable-pytest-warnings
```
Comment (blueyed) [inline setup.cfg]: Booh!
Rather fix or ignore them.

File: setup.cfg [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [tool:pytest]
 addopts = --tb=short --ds=carbondoomsday.settings
-testpaths = carbondoomsday/carbondioxide/tests
+testpaths = ./carbondoomsday/carbondioxide/test/
```
Comment (blueyed) [inline setup.cfg]: the `./` should not be necessary.

File: carbondoomsday/carbondioxide/models.py [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
 
 from django.db import models
+
+
+# FIXME
+# Should add some way to make date unique
+# Add tests
+# Add migration
+class CO2Measurement(models.Model):
+    date = models.DateField()
```
Comment (blueyed) [inline carbondoomsday/carbondioxide/models.py]: Make date unique?!

File: carbondoomsday/carbondioxide/tasks.py [hunk 4]
```diff
@@ -1,8 +1,37 @@
...
+import requests
+
 from carbondoomsday.celery import app
 
 
 @app.task
-def scrape_last_five_days():
-    print("Scrape that data!")
+def scrape_latest_co2_data():
+    # from carbondoomsday.carbondioxide.models import CO2Measurement
```
Comment (blueyed) [inline carbondoomsday/carbondioxide/tasks.py]: left-over comment.

File: carbondoomsday/carbondioxide/tasks.py [hunk 5]
```diff
@@ -1,8 +1,37 @@
...
+
+    url = "https://www.esrl.noaa.gov/gmd/webdata/ccgg/trends/co2_mlo_weekly.csv"  # noqa
```
Comment (blueyed) [inline carbondoomsday/carbondioxide/tasks.py]: Move to settings?!

File: carbondoomsday/carbondioxide/tasks.py [hunk 6]
```diff
@@ -1,8 +1,37 @@
...
+
+    try:
+        response = requests.get(url)
+    except Exception as err:
+        logging.error("Failed to retrieve CSV from '{}'.")
+        raise(err)
+
+    decoded = str(response.content, "utf-8")
+    separated = decoded.split("\n")
```
Comment (blueyed) [inline carbondoomsday/carbondioxide/tasks.py]: Single quotes are nicer.. ;)

File: carbondoomsday/carbondioxide/test/test_tasks.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+def test_scrape_latest_parse_failure():
+    pass
+
+
+def test_scrape_latest_existing_models():
+    pass
+
+
+def test_scrape_latest_missing_data():
+    pass
```
Comment (blueyed) [inline carbondoomsday/carbondioxide/test/test_tasks.py]: Better not to have just-passing tests.

File: setup.cfg [hunk 8]
```diff
@@ -1,3 +1,3 @@
 [tool:pytest]
-addopts = --tb=short --ds=carbondoomsday.settings
-testpaths = carbondoomsday/carbondioxide/tests
+addopts = --tb=short --ds=carbondoomsday.settings --disable-pytest-warnings
+testpaths = ./carbondoomsday/carbondioxide/test/
```
Comment (blueyed) [inline setup.cfg]: Remove "./" here, too?!
Also `--disable-pytest-warnings`.
(Probably commented on the commit before)

Comment (decentral1se) [reply setup.cfg]: > Also --disable-pytest-warnings. 

Ah, but how do I then disable annoying 3rd party warnings?

**Comment by decentral1se** [comment]
> btw: no punctuation with commit log summaries is nicer.

At some point, I thought - 'why do I not write grammatically correct sentences here!?' So I started :smile:

File: setup.cfg (hunk 8 above)
Comment (blueyed) [reply setup.cfg]: Fix them there.. ;)
Related: https://github.com/pytest-dev/pytest/issues/2449 - with that you could filter them with logging filters.

**Pull request merged**